biocypher_metta/adapters/gaf_adapter.py

```diff
@@ -73,32 +95,66 @@ def load_rnacentral_mapping(self):
                 self.rnacentral_mapping[mapping[0] +
                                         '_' + mapping[3]] = mapping[2]
 
-    def get_edges(self):
+    def parse_qualifier(self, qualifier):
+        """Parse the qualifier to detect negation and return the negated status."""
+        negated = False
+        if "NOT" in qualifier:
+            negated = True
+        return negated
 
+    def get_edges(self):
         if self.type == 'rna':
             self.load_rnacentral_mapping()
 
         with gzip.open(self.filepath, 'rt') as input_file:
             for annotation in gafiterator(input_file):
-                source = annotation['GO_ID']
-                target = annotation['DB_Object_ID']
+                source = annotation['DB_Object_ID']
+                target = annotation['GO_ID']
 
+                # Skip if subontology doesn't match
+                if self.subontology and self.subontology_mapping:
+                    go_subontology = self.subontology_mapping.get(target)
+                    if go_subontology != self.subontology:
+                        continue
+
+                # RNA-specific mapping
                 if self.type == 'rna':
-                    transcript_id = self.rnacentral_mapping.get(
-                        annotation['DB_Object_ID'])
+                    transcript_id = self.rnacentral_mapping.get(annotation['DB_Object_ID'])
                     if transcript_id is None:
                         continue
-                    target = transcript_id
+                    source = transcript_id
+
+                # Cellular component filtering using qualifier
+                qualifier = annotation['Qualifier']
+                if self.label.startswith('cellular_component_gene_product'):
+                    if 'part_of' in qualifier:
+                        if 'part_of' not in self.label:
+                            continue
+                    elif 'located_in' in qualifier:
+                        if 'located_in' not in self.label:
+                            continue
+                    else:
+                        continue
+
+                # Determine negation from qualifier
+                negated = self.parse_qualifier(qualifier)
+
                 props = {}
                 if self.write_properties:
                     props = {
-                        'qualifier': annotation['Qualifier'],
+                        'qualifier': qualifier,
                         'db_reference': annotation['DB:Reference'],
-                        'evidence': annotation['Evidence']
+                        'evidence': annotation['Evidence'],
+                        'negated': str(negated).lower()  
```

Comment:
What's does the `negated` property signify? Can you add few lines of comment explaining it?